tests/rptest/tests/upgrade_test.py

```diff
@@ -245,19 +251,36 @@ def test_upgrade_with_all_workloads(self, single_upgrade):
             elif current_version[0:2] == (22, 1) and paused:
                 self._producer.start(clean=False)
 
+            max_retries = 5
             # Wait for consumers to finish work and check invariants.
-            for consumer in self._consumers:
-                consumer.wait()
+            while max_retries > 0:
+                max_retries -= 1
+                for consumer in self._consumers:
+                    try:
+                        consumer.wait()
+                        break
```

Comment:
Not sure I'm reading this correctly - why are we breaking out of the `_consumers` loop if there is no exception? Should we break out of the retries loop instead?

Reply:
exactly